File: Factory Method Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Factory Method Pattern in Java");
        System.out.println("------------------------------");
        System.out.println();
  
        AbstractCreator creator1 = new ConcreteCreator1();
        AbstractCreator creator2 = new ConcreteCreator2();

        AbstractProduct productA1 = creator1.createProduct();
        AbstractProduct productA2 = creator2.createProduct();

        productA1.doSomething();
        productA2.doSomething();
    }
}

File: Factory Method Pattern/1. Minimal/Python/main.py
from concrete_creator_1 import ConcreteCreator1
from concrete_creator_2 import ConcreteCreator2

def main():
    print("Factory Method Pattern in Python")
    print("--------------------------------")
    print()

    creator1 = ConcreteCreator1()
    creator2 = ConcreteCreator2()

    productA1 = creator1.createProduct()
    productA2 = creator2.createProduct()

    productA1.doSomething()
    productA2.doSomething()

if __name__ == "__main__":
    main()
